.github/workflows/release.yaml

```diff
@@ -0,0 +1,69 @@
+# When pushing to main, we run checks and then release.
+
+name: Check and Release
+on:
+  push:
+    branches:
+    - main
+permissions:
+  contents: read
+env:
+  GENERATED_PACKAGE_DIR: dist
+  ARTIFACT_NAME: artifact-ubuntu-latest-3.10
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yaml
+    permissions:
+      contents: write
+    secrets: inherit
+
+  provenance:
+    needs: build
+    permissions:
+      actions: read
+      id-token: write
+      contents: read
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@d995948e8d53cc639c0d3ef69db31dbc243519c4
+    with:
+      base64-subjects: ${{ needs.build.outputs.artifacts-sha256 }}
+
+  release:
+    if: "!startsWith(github.event.commits[0].message, 'bump:')"
+    needs: [build, provenance]
+    name: Release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # To publish release notes.
+    steps:
+    - name: Harden Runner
+      uses: step-security/harden-runner@74b568e8591fbb3115c70f3436a0c6b0909a8504
+      with:
+        egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+    - name: Checkout
+      uses: actions/checkout@d0651293c4a5a52e711f25b41b05b2212f385d28
+      with:
+        fetch-depth: 0
+        ref: main
+    - name: Download provenance
+      uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741
+      with:
+        name: ${{needs.provenance.outputs.attestation-name}}
+    - name: Download artifact
+      uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741
+      with:
+        name: ${{ env.ARTIFACT_NAME }}
+        path: ${{ env.GENERATED_PACKAGE_DIR }}
+    - name: Verify the artifact hash
+      env:
+        ARTIFACT_HASH: ${{ needs.build.outputs.artifacts-sha256 }}
+      run: |
+        set -euo pipefail
+        echo "hash of package should be ${{ env.ARTIFACT_HASH }}"
+        # Verify hashes.
+        echo "${{ env.ARTIFACT_HASH }}" | base64 -d | sha256sum --strict --check --status || exit -2
+    - name: Upload assets
+      env:
+        GH_TOKEN: ${{ secrets.REPO_ACCESS_TOKEN }}
+      run: |
+        gh release create `git describe --tags --abbrev=0` ./${{ env.GENERATED_PACKAGE_DIR }}/* ${{needs.provenance.outputs.attestation-name}} -F CHANGELOG.md
```

Comment:
Can we please use `--notes-file` instead of `-F`? It’s more descriptive.

It’s not a draft release, but will publish the release? Which in turn will trigger the [`release` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) with `"published"` type?

Reply:
> Can we please use `--notes-file` instead of `-F`? It’s more descriptive.

Yes!

Reply:
> It’s not a draft release, but will publish the release? Which in turn will trigger the [`release` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) with `"published"` type?

Yes, I think so.